src/Common/Handlers/InfrastructureUtilities.cs

```diff
@@ -0,0 +1,185 @@
+        /// <param name="force">Force variable that indicates if the program can continue if the Customer Infrastructure already exists.</param>
+        /// <param name="customerInfrastructureName">Name of Customer Infrastructure</param>
+        /// <returns></returns>
+        public static async Task<CustomerInfrastructure> CheckCustomerInfrastructureAlreadyExists(ISession session, ICustomerPortalClient customerPortalClient, bool force, string customerInfrastructureName)
+        {
+            CustomerInfrastructure currentCustomerInfrastructure = await GetCustomerInfrastructure(session, customerPortalClient, customerInfrastructureName, false);
+
+            // if customerInfrastructure doesn't exists. Can continue to be created.
+            // else check the force value
+
+            if (currentCustomerInfrastructure != null)
+            {
+                if (force)
+                {
+                    session.LogInformation($"The Customer Infrastructure with name '{customerInfrastructureName}' already exists but the force flag has been set. Continuing to create new version.");
+                }
+                else
+                {
+                    string errorMessage = $"The Customer Infrastructure with name '{customerInfrastructureName}' already exists and cannot be created. If you want to continue, please use the 'Force' flag on the command.";
+                    session.LogError(errorMessage);
+                    throw new Exception(errorMessage);
+                }
+            }
+
+            return currentCustomerInfrastructure;
+        }
+
+        /// <summary>
+        /// Get Customer Infrastructure
+        /// </summary>
+        /// <param name="session">session</param>
+        /// <param name="customerPortalClient">customer portal client</param>
+        /// <param name="customerInfrastructureName">customer infrastructure name</param>
+        /// <param name="throwIfNotExits">throw and exception if not exists</param>
+        /// <returns></returns>
+        public static async Task<CustomerInfrastructure> GetCustomerInfrastructure(ISession session, ICustomerPortalClient customerPortalClient, string customerInfrastructureName, bool throwIfNotExits)
+        {
+            CustomerInfrastructure currentCustomerInfrastructure = null;
+            try
+            {
+                session.LogInformation($"Checking if exists the current Customer Infrastructure with name '{customerInfrastructureName}'.");
+                currentCustomerInfrastructure = await customerPortalClient.GetObjectByName<CustomerInfrastructure>(customerInfrastructureName);
+                session.LogInformation($"Customer Infrastructure with name '{customerInfrastructureName}' exists.");
+            }
+            catch (CmfFaultException ex) when (ex.Code?.Name == Cmf.Foundation.Common.CmfExceptionType.Db20001.ToString() && ex.Message.Contains("was not found in the system"))
```

Comment:
I don't think that asserting that bit of the message is needed since the proper way is to verify the code. It's also not future proof since that message can change.

Reply:
Hmmm I asked myself when when I compare with the code. I don't like to compare message or string because what you say, but my doubt is if the code will be enough ^^ that is my fear. Or if this code may appear for other situations that do not "already exist"

Reply:
Yes, the framework has a code for each type of exception :)